runtime/shared_heap.c

```diff
@@ -209,7 +209,8 @@ static void calc_pool_stats(pool* a, sizeclass sz, struct heap_stats* s) {
   s->pool_frag_words += Wsize_bsize(POOL_HEADER_SZ);
 
   while (p + wh <= end) {
-    header_t hd = (header_t)*p;
+    header_t hd = (header_t)atomic_load_explicit((atomic_uint*)p,
```

Comment:
I strongly suspect you meant `atomic_uintnat`.
```suggestion
    header_t hd = (header_t)atomic_load_explicit((atomic_uintnat*)p,
```